reasoner/resolution/resolver/ConcludableResolver.java

```diff
@@ -162,37 +163,44 @@ protected void initialiseDownstreamActors() {
     }
 
     @Override
-    protected ResponseProducer responseProducerCreate(Request request, int iteration) {
-        LOG.debug("{}: Creating a new ResponseProducer for request: {}", name(), request);
-        Actor<RootResolver> root = request.path().root();
+    protected ResponseProducer responseProducerCreate(Request fromUpstream, int iteration) {
+        LOG.debug("{}: Creating a new ResponseProducer for request: {}", name(), fromUpstream);
+        Actor<RootResolver> root = fromUpstream.path().root();
         iterationStates.putIfAbsent(root, new IterationState(iteration));
         IterationState iterationState = iterationStates.get(root);
 
-        Traversal traversal = boundTraversal(concludable.conjunction().traversal(), request.partialAnswer().conceptMap());
-        ResourceIterator<ConceptMap> traversalProducer = traversalEngine.iterator(traversal).map(conceptMgr::conceptMap);
+        Traversal traversal = boundTraversal(concludable.conjunction().traversal(), fromUpstream.partialAnswer().conceptMap());
+        assert fromUpstream.partialAnswer().isMapped();
+        ResourceIterator<UpstreamVars.Derived> upstreamAnswers = traversalEngine.iterator(traversal)
+                .map(conceptMgr::conceptMap)
+                .map(conceptMap -> fromUpstream.partialAnswer().asMapped().mapToUpstream(conceptMap));
```

Comment:
when we save a traversal to the `ResponseProducer`, we first map it through to a `Derived` answer state